src/samples/coap/lwm2m-client.c

```diff
@@ -1263,6 +1250,46 @@ main(int argc, char *argv[])
         return -1;
     }
 
+    switch (sec_mode) {
+    case SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY:
+        if (data_ctx.is_bootstrap) {
+            public_key_or_id = sol_blob_new_dup(CLIENT_BS_PSK_ID,
+                sizeof(CLIENT_BS_PSK_ID) - 1);
+            secret_key = sol_blob_new_dup(CLIENT_BS_PSK_KEY, PSK_KEY_LEN);
+        } else {
+            public_key_or_id = sol_blob_new_dup(CLIENT_SERVER_PSK_ID,
+                sizeof(CLIENT_SERVER_PSK_ID) - 1);
+            secret_key = sol_blob_new_dup(CLIENT_SERVER_PSK_KEY, PSK_KEY_LEN);
+        }
+        server_public_key = NULL;
+        break;
+    case SOL_LWM2M_SECURITY_MODE_RAW_PUBLIC_KEY:
+        r = sol_util_base16_decode(buf_aux, sizeof(buf_aux),
+            sol_str_slice_from_str(CLIENT_PRIVATE_KEY), SOL_DECODE_BOTH);
+        secret_key = sol_blob_new_dup(buf_aux, RPK_PRIVATE_KEY_LEN);
+        r = sol_util_base16_decode(buf_aux, sizeof(buf_aux),
+            sol_str_slice_from_str(CLIENT_PUBLIC_KEY), SOL_DECODE_BOTH);
+        public_key_or_id = sol_blob_new_dup(buf_aux, RPK_PUBLIC_KEY_LEN);
+
+        if (data_ctx.is_bootstrap) {
+            r = sol_util_base16_decode(buf_aux, sizeof(buf_aux),
+                sol_str_slice_from_str(BS_SERVER_PUBLIC_KEY), SOL_DECODE_BOTH);
+            server_public_key = sol_blob_new_dup(buf_aux, RPK_PUBLIC_KEY_LEN);
+        } else {
+            r = sol_util_base16_decode(buf_aux, sizeof(buf_aux),
+                sol_str_slice_from_str(SERVER_PUBLIC_KEY), SOL_DECODE_BOTH);
+            server_public_key = sol_blob_new_dup(buf_aux, RPK_PUBLIC_KEY_LEN);
+        }
+        break;
+    case SOL_LWM2M_SECURITY_MODE_CERTIFICATE:
```

Comment:
I left it like this so we can go through the sample code itself and only fail on a "Certificate Mode Not Supported" at the actual APIs; kind of a test. What do you think?